src/bucket/test/BucketManagerTests.cpp

```diff
@@ -327,6 +327,99 @@ TEST_CASE("bucketmanager missing buckets fail", "[bucket][bucketmanager]")
     }
 }
 
+#ifdef ENABLE_NEXT_PROTOCOL_VERSION_UNSAFE_FOR_PRODUCTION
+TEST_CASE_VERSIONS("bucketmanager snapshots BucketList size",
+                   "[bucket][bucketmanager]")
+{
+    VirtualClock clock;
+    Config cfg(getTestConfig(0, Config::TESTDB_IN_MEMORY_SQLITE));
+    cfg.USE_CONFIG_FOR_GENESIS = true;
+
+    auto app = createTestApplication<BucketTestApplication>(clock, cfg);
+    LedgerManagerForBucketTests& lm = app->getLedgerManager();
+    BucketManagerImpl& bm =
+        dynamic_cast<BucketManagerImpl&>(app->getBucketManager());
+
+    uint32_t windowSize;
+    std::deque<uint64_t> correctWindow;
+    {
+        LedgerTxn ltx(app->getLedgerTxnRoot());
+        windowSize = app->getLedgerManager()
+                         .getSorobanNetworkConfig(ltx)
+                         .stateExpirationSettings()
+                         .bucketListSizeWindowSampleSize;
+        for (auto i = 0; i < windowSize; ++i)
+        {
+            correctWindow.push_back(0);
+        }
+
+        ltx.commit();
+    }
+
+    auto check = [&]() {
+        // Check in-memory average from BucketManager
+        uint64_t sum = 0;
+        for (auto e : correctWindow)
+        {
+            sum += e;
+        }
+
+        uint64_t correctAverage = sum / correctWindow.size();
+
+        LedgerTxn ltx(app->getLedgerTxnRoot(), false,
+                      TransactionMode::READ_ONLY_WITHOUT_SQL_TXN);
+        REQUIRE(bm.getAverageBucketListSize(ltx) == correctAverage);
+
+        // Check on-disk sliding window
+        LedgerKey key(CONFIG_SETTING);
+        key.configSetting().configSettingID =
+            ConfigSettingID::CONFIG_SETTING_BUCKETLIST_SIZE_WINDOW;
+        auto txle = ltx.loadWithoutRecord(key, /*loadExpiredEntry=*/false);
+        releaseAssert(txle);
+        auto const& leVector =
+            txle.current().data.configSetting().bucketListSizeWindow();
+
+        REQUIRE(correctWindow.size() == leVector.size());
```

Comment:
Can we just use `REQUIRE(correctWindow == leVector)` ? (you may convert deque to a vec for this to work)